File: Unit2-06/Jet.ts
/*
* This is the standard jet class.
*
* @author  Infinity de Guzman
* @version 1.0
* @since   2024-06-08
*/

import { Airplane } from "./Airplane"

/**
* This is the jet class.
*/
export class Jet extends Airplane {
    /** Jet speed change multiplier. */
    private MULTIPLIER: number = 2

    // Constructor.
    public Jet() {
        // super()
    }

    // Set jet speed.
    public setSpeed(speed: number) {
        super.setSpeed(speed * this.MULTIPLIER)
    }

    // Double speed.
    public accelerate() {
        super.setSpeed(super.getSpeed() * 2)
    }
}

File: Unit2-06/java/Jet.java
/*
* This is the standard jet class.
*
* @author  Infinity de Guzman
* @version 1.0
* @since   2024-06-08
*/

/**
* This is the jet class.
*/
public class Jet extends Airplane {
    /** Jet speed change multiplier. */
    private static final int MULTIPLIER = 2;

    /**
    * Constructor.
    */
    public Jet() {
        // Default constructor
    }

    /**
    * Set jet speed.
    *
    * @param speed set speed
    */
    @Override
    public void setSpeed(int speed) {
        super.setSpeed(speed * MULTIPLIER);
    }

    /**
    * Double speed.
    */
    public void accelerate() {
        super.setSpeed(super.getSpeed() * 2);
    }
}
